geomesa-utils/src/main/scala/org/locationtech/geomesa/utils/index/ByteArrays.scala

```diff
@@ -169,6 +225,25 @@ object ByteArrays {
     }
   }
 
+  def concat(first: Array[Byte], second: Array[Byte]): Array[Byte] = {
```

Comment:
this should hopefully be slightly more efficient - `++` will use an implicit conversion to GenTraversable and then an Array.newBuilder, which allocates the array and then copies values in, so saves a few intermediate steps but doesn't do anything egregious.
my main impetus was to replace guava Bytes.concat calls, so that we can one day not depend on guava at all (i can dream, right?)

Reply:
+1 yeah there's likely some other places like that we could optimize. I definitely think minimizing guava is good...or at least shading it all the time...but for something that simple no need to introduce guava.